spec/index/multi_index_spec.rb

```diff
@@ -35,6 +35,30 @@
           labels: [[0,0,1,1,2,2]])
       }.to raise_error
     end
+
+    context "create an MultiIndex with name" do
+      context "if no name is set" do
+        let(:mi) { Daru::MultiIndex.new(
+                    levels: [[:a,:b,:c], [:one, :two]],
+                    labels: [[0,0,1,1,2,2], [0,1,0,1,0,1]]) }
+        it { expect(mi.name).to be_nil }
+      end
+
+      context "correctly return the MultiIndex name" do
+        let(:mi) { Daru::MultiIndex.new(
+                  levels: [[:a,:b,:c], [:one, :two]],
+                  labels: [[0,0,1,1,2,2], [0,1,0,1,0,1]], name: ['n1', 'n2']) }
+        it { expect(mi.name).to eq ['n1', 'n2'] }
```

Comment:
What if number of names are different from number of levels? We need a test on this (assuring there would be readable error message).